modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js

```diff
@@ -69,14 +69,28 @@ const useFormSubmit = ({apiRef, containerRef}) => {
 							return;
 						}
 
-						Liferay.Util.submitForm(event.target);
+						const avoidSubmit =
+							event.target?.dataset.ddmFormAvoidSubmit;
+
+						if (!avoidSubmit) {
+							Liferay.Util.submitForm(event.target);
+						}
+
+						Liferay.fire('ddmFormValid', {
+							formId: event.target.id,
+						});
 
 						Liferay.fire('ddmFormSubmit', {
 							formId: getFormId(
 								getFormNode(containerRef.current)
 							),
 						});
 					}
+					else {
+						Liferay.fire('ddmFormError', {
+							formId: event.target.id,
```

Comment:
Would using the `getFormId` return the same value?

Reply:
Same reasons, the `getFormId` is returning undefined and I not sure if this id is the dom id of the wrapper form.